### PR TITLE
WFCORE-5037 - Handle maps of complex objects in JMX Facade

### PR DESCRIPTION
**Pull request by kabir**
https://issues.redhat.com/browse/WFCORE-5037

### CODE REVIEW

**Comment by darranl** [comment]
@jmesnil I think you were talking about this one last week, were there still issues you thought should be addressed?

File: controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java [hunk 1]
```diff
@@ -1295,6 +1295,14 @@ public ModelNode getUndefinedMetricValue() {
         return undefinedMetricValue;
     }
 
+    /**
+     * @deprecated internal use only
+     */
+    @Deprecated
+    public boolean isNestedMap() {
```
Comment (jmesnil) [inline controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java:1302]: As we discussed during the meeting, I don't think that method belong to the `AttributeDefinition` API. You mentioned that we can achieve the same by doing an `instanceof` check for the `ObjectMapAttributeDefinition` in https://github.com/wildfly/wildfly-core/pull/4260/files#diff-738bea9b01b7d11ea0c7968b4390eeccR299.

**Comment by kabir** [comment]
@jmesnil I have removed that deprecated method and am using instanceof now

**Comment by jmesnil** [comment]
@kabir thanks

**Pull request merged**